skimage/graph/_graph_cut.py

```diff
@@ -267,17 +267,18 @@ def _ncut_relabel(rag, thresh, num_cuts, random_generator):
     d, w = _ncut.DW_matrices(rag)
     m = w.shape[0]
 
-    if m > 2:
-        if (d != w).nnz == 0:
-            # Degenerate case, usually 3 single pixels.
-            #
-            # We're not sure exactly why this case arises. For SciPy <= 0.14, SciPy continued
-            # to compute an eigenvector, but later versions (correctly) won't.
-            # We refuse to guess, and give up.
-            #
-            # It may make sense to a warning here; on the other hand segmentations
-            # are not a ground truth, so this level of "noise" should be acceptable.
-            return
+    if (m > 2) and (d != w).nnz > 0:
+        # This avoids further segmenting a graph that is too small,
+        # and the degenerate case (d == w), which typically occurs
+        # when only three single pixels remain.
+        #
+        # We're not sure exactly why this latter case arises. For
+        # SciPy <= 0.14, SciPy continued to compute an eigenvector,
+        # but newer versions (correctly) won't.  We refuse to guess,
+        # and stop further segmentation.
+        #
+        # It may make sense to a warning here; on the other hand segmentations
+        # are not a ground truth, so this level of "noise" should be acceptable.
```

Comment:
I might be confused here but I think the `_label_all(rag, 'ncut label')` should be indented as well and the comment above updated? Otherwise it's confusing because the body of the if-statement now executes for the **non**-degenerate case?

Reply:
The body of the if statement runs for the non-degenerate case, yes. The comment is to explain why the if-clause is necessary in the first place. We could move it around, to make it clearer.